### ISSUE TITLE
[UX] Reintroduce the (removed in 1.5.0) Login block (either as an AJAXy drop-down menu or pop-up dialog.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2207]
Follow-up to #2163...

@quicksketch [said](https://github.com/backdrop/backdrop-issues/issues/2163#issuecomment-246530890):

> As @jenlampton brought up to me, we're going to have an issue that our menu links don't have the ability to redirect back to the page you came from. So this means for now at least, clicking login will actually take you away from the page.

> As an alternative, we could actually make the login link open a dialog (via AJAX) with the login form inside of it. But how we would enable/disable that option isn't figured out at this point.

FYI: our recently added [SmartMenus library](http://www.smartmenus.org) (that handles core drop-down menus) also supports mega-menus:

![smartmenus-mega-menu](https://cloud.githubusercontent.com/assets/2423362/18615138/dce434c2-7de1-11e6-87db-103e4bb5c62a.png)

...not suggesting that this is the best way to go about it. Just a thought, because that way:

- we make good use of a library already available in core (and if I am not wrong, already loaded on the page as the primary navigation is drop-down enabled by default in >1.5.0)
- we gain all the usability features that this library brings (keyboard navigation for example)

**Comment by jenlampton** [comment]
The login block is still there, we just removed it from the sidebar.

**Comment by klonos** [comment]
Yes, I know that the block itself was not removed from core, just from the default layout and also not included in the new home page layout. Now it exists as a "login" link in the account menu, but that effectively requires an extra click and page load in order to log in. And AFAICT, that doesn't redirect you to the page you were viewing. It takes you to your user account page instead.

There's a few ways to tackle this:

- add a destination to the login link, so that the user gets redirected back to the page they clicked the login link from.
- make the link pop up an AJAX dialog that exposes a username/password set of fields and logs people in without loading the entire login page.
- instead of a pop up dialog, we use a mega-menu that embeds the username/password fields.

**Comment by jenlampton** [comment]
> add a ?destination= to the login link

I think the menu cache prevents this from being updated on every page load. We may need to do add the destination via JS if that's the case.  I like both alternate solutions for this, but I think a dialog might be easier to implement than the mega-menu feature. 

In either case, I'm not sure this is the type of bug we need to fix asap, so removing the 1.5.1 milestone. (all bugs can be put in at any time)
